src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```diff
@@ -56,24 +55,55 @@ public DebtPositionDTO saveDebtPosition(DebtPositionDTO debtPositionDTO, Organiz
       savedPaymentOption.getInstallments().forEach(installmentNoPII -> {
         Installment mappedInstallment = mappedDebtPosition.getSecond().get(installmentNoPII);
         mappedInstallment.setPaymentOptionId(savedPaymentOption.getPaymentOptionId());
-        if (StringUtils.isBlank(mappedInstallment.getIud())) {
-          String iud = Utilities.getRandomIUD();
-          mappedInstallment.setIud(iud);
-          installmentNoPII.setIud(iud);
-        }
+        checkInstallment(debtPositionTypeOrg, mappedInstallment, installmentNoPII, org);
+
         InstallmentNoPII savedInstallment = installmentRepository.save(mappedInstallment).getNoPII();
         installmentNoPII.setPersonalDataId(savedInstallment.getPersonalDataId());
         installmentNoPII.setInstallmentId(savedInstallment.getInstallmentId());
         installmentNoPII.setPaymentOptionId(savedPaymentOption.getPaymentOptionId());
 
         mappedInstallment.getTransfers().forEach(transfer -> {
           transfer.setInstallmentId(savedInstallment.getInstallmentId());
+          checkTransfer(debtPositionTypeOrg, transfer, org);
           transferRepository.save(transfer);
         });
       });
     });
 
     return debtPositionMapper.mapToDto(savedDebtPosition);
   }
+
+  private void checkDebtPosition(DebtPositionDTO debtPositionDTO, Pair<DebtPosition, Map<InstallmentNoPII, Installment>> mappedDebtPosition, Organization org) {
```

Comment:
why this check? this API could be called for update use case...
anyway, during creation, there is the checkUniqueness... isn't it verifying this ids?
During creation there is also the IUV generation... the  IUPD generation should be performed also there

ps: you will go into conflict with @Benedetta-fabbri task